jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala

```diff
@@ -105,6 +116,7 @@ final case class JmsBrowseSettings(connectionFactory: ConnectionFactory,
     extends JmsSettings {
   def withCredential(credentials: Credentials): JmsBrowseSettings = copy(credentials = Some(credentials))
   def withQueue(name: String): JmsBrowseSettings = copy(destination = Some(Queue(name)))
+  def withDestination(destination: Queue): JmsBrowseSettings = copy(destination = Some(destination))
```

Comment:
This should allow any `Destination` sub-type.

Reply:
The reason is  JmsBrowseSettings restrict its destination as Queue. I guess that's intended to distinguish with JmsConsumerSettings


final case class JmsBrowseSettings(connectionFactory: ConnectionFactory,
                                   destination: Option[Queue] = None,

two options:

1 keep existing JmsBrowseSettings working with Queue only, remove the final from 
final case class Queue(override val name: String) extends Destination
to enable passing custom Queue to JmsBrowseSettings 

2 Change JmsBrowseSettings to work with Destination, But that will defeat the purpose of this class.


I will use option1 for now. Let me know your comments please.